### ISSUE TITLE
One-liner to install release package on Debian 12

### ISSUE THREAD

**Issue by ComputaBloke** [cockpit-project/cockpit-files#635]
I have been hanging out for this to hit general release. Thanks for your efforts.
I noticed the release isn't in the Cockpit 'Applications' (appstream) list yet, nor on most distribution repos until it can be backported.

I wanted to share a quick install script (without all the Make/developer dependencies).

```
wget https://github.com/cockpit-project/cockpit-files/releases/latest/download/cockpit-files-2.tar.xz && \
  tar -xf cockpit-files-2.tar.xz cockpit-files/dist && \
  mv cockpit-files/dist /usr/share/cockpit/files && \
  rm -r cockpit-files && \
  rm cockpit-files-2.tar.xz
  ```
_Note: the '-2' I expect might be best removed in future releases_

This works for me on Debian GNU/Linux 12 (bookworm) where rest of cockpit-project was installed via standard `apt install ...`

If you're able to add this to the README.md I'm sure others will find it useful.
Thanks again

**Comment by jelly** [comment]
Thanks for the issue, in general we don't want to recommend putting files into package manager owned directories such as `/usr/share/cockpit`. We do recommend using `make devel-install`  as this installs to ~/.local/share/cockpit so easily reverted by a normal user.

For more detail see https://cockpit-project.org/guide/latest/packages.html#package-layout

**Comment by martinpitt** [comment]
Also, please don't run `wget` as root. You can trim this a bit to use pipes, like `curl https://... | sudo tar ...` and please install into /usr/local/share/cockpit to avoid conflicts with the package manager.

I just sent https://bugs.debian.org/1076001 to start the inclusion of cockpit-files into Debian. Thanks for the nudge :wink:

**Comment by emaayan** [comment]
> Also, please don't run `wget` as root. You can trim this a bit to use pipes, like `curl https://... | sudo tar ...` and please install into /usr/local/share/cockpit to avoid conflicts with the package manager.
> 
> I just sent [bugs.debian.org/1076001](https://bugs.debian.org/1076001) to start the inclusion of cockpit-files into Debian. Thanks for the nudge 😉

so this would be the reason i can't use apt install cockpit-files on ubuntu 24 too?

**Comment by martinpitt** [comment]
Just for the record, it's been sitting in the [NEW queue](https://ftp-master.debian.org/new.html) for a week.

**Comment by machsix** [comment]
I don't think it works. The latest version cockpit in Debian is [287.1-0](https://packages.debian.org/bookworm/cockpit). But the cockpit-files requires ` "cockpit": "318"`. You can check https://salsa.debian.org/debian/cockpit-files/-/blob/master/dist/manifest.json?ref_type=heads or this repo for file `dist/manifest.json`.

If I copy the dist folder,  here is what I get
![Screenshot from 2024-08-13 10-14-24](https://github.com/user-attachments/assets/cbf76765-fedc-4122-b1a2-c4ea00c96bbe)

**Comment by martinpitt** [comment]
@machsix stable-backports has version 322. See https://cockpit-project.org/running.html#debian

**Comment by machsix** [comment]
Thanks! I just recently switched from Arch to Debian.. kind of newbiee. After running `apt install -t bookworm-backports cockpit`, it works!

**Comment by emaayan** [comment]
So when it will available offcially for ubuntu 24 lts?

**Comment by martinpitt** [comment]
I can't say. It needs to be accepted into Debian (that's the big unknown), then synced into the current Ubuntu dev release (that's quick, I can do it), then reviewed by the Ubuntu backports team (which takes roughly a week).

**Comment by neitsab** [comment]
Meanwhile, reformulating @ComputaBloke instructions while taking into account the devs' advice/requests, I ended up with the following:

```shell
cd /tmp/
wget https://github.com/cockpit-project/cockpit-files/releases/download/9/cockpit-files-9.tar.xz -qO- | tar -xJ
# for global install
sudo install cockpit-files/dist/* -Dt /usr/local/share/cockpit/files/
# for local one
install cockpit-files/dist/* -Dt ~/.local/share/cockpit/files/
```

After a quick check with `cockpit-bridge --packages`, the File Browser app was indeed available in the interface :+1:

**Comment by B3rn475** [comment]
Any update?
It looks like the request has been sitting there for 6 months.

Would it be possible to have an official ".deb" file to manually install? Instead of having to copy files around?

**Comment by martinpitt** [comment]
Indeed it's hugely annoying that the NEW queue processing has stalled for that long, it never was that bad. In the meantime I uploaded a current version to https://launchpad.net/~pitti/+archive/ubuntu/cockpit-files . It says "ubuntu noble", but it's an Architecture: all package. So you should be able to add that PPA to your apt sources, or just [grab the deb manually](https://launchpad.net/~pitti/+archive/ubuntu/cockpit-files/+packages)

**Comment by m-primo** [comment]
> Indeed it's hugely annoying that the NEW queue processing has stalled for that long, it never was that bad. In the meantime I uploaded a current version to https://launchpad.net/~pitti/+archive/ubuntu/cockpit-files . It says "ubuntu noble", but it's an Architecture: all package. So you should be able to add that PPA to your apt sources, or just [grab the deb manually](https://launchpad.net/~pitti/+archive/ubuntu/cockpit-files/+packages)

Thank you!
It worked on Debian 12.
I downloaded the .deb package `wget https://launchpad.net/~pitti/+archive/ubuntu/cockpit-files/+files/cockpit-files_14-0ppa1_all.deb`, then installed it `apt install ./cockpit-files_14-0ppa1_all.deb`, and restarted the cockpit service and it worked.

**Comment by martinpitt** [comment]
The package finally was reviewed in Debian, unfortunately rejected: https://alioth-lists.debian.net/pipermail/pkg-utopia-maintainers/2025-April/036800.html That's a setback, I'll discuss this with the ftpmasters and hopefully we'll find a solution.

**Comment by emaayan** [comment]
why was it rejected?

**Comment by GeistFighter** [comment]
rejected ? why ? 😥

**Comment by martinpitt** [comment]
Here's hope that this time around the review will be faster.. 🤞

**Comment by martinpitt** [comment]
@B3rn475 The package is not architecture specific, you can use it on arm64 already from the PPA.

**Comment by martinpitt** [comment]
Once it propagates to testing, I'll upload it to trixie-backports.

**Comment by B3rn475** [comment]
> [@B3rn475](https://github.com/B3rn475) The package is not architecture specific, you can use it on arm64 already from the PPA.

Sorry for the confusion.
You are right the Debian package is for platform `all`.

I assumed the Debian and Ubuntu packages were configured in the same way.
I tried recently on a Raspberry Pi running Ubuntu and it complained about not being compatible with `arm64` because it is only available for `amd64`.
https://launchpad.net/~pitti/+archive/ubuntu/cockpit-files/+packages

As I moved back to RaspberryOS (Debian) I should be OK after it propagated to `trixie-backports`.
I guess there is something that could be improved on the Ubuntu side of the package.
I may have made some mistake though, feel free to correct me.

**Comment by martinpitt** [comment]
Ah, maybe PPAs report it that way, sorry. You can download and install https://launchpad.net/~pitti/+archive/ubuntu/cockpit-files/+files/cockpit-files_35-0ppa1_all.deb directly.
